ocaml/testsuite/tests/mixed-blocks/hash.ml

```diff
@@ -18,13 +18,20 @@ all_modules = "hash.ml"
 
 (* Currently bytecode/native hashes of mixed records are different.
    Mixed records are represented as mixed blocks in native code and
-   normal blocks in bytecode. We don't make any special effort to get
-   their hash values to line up. This is something we could consider
-   revisiting -- the simplest way to accomplish this may well be to
-   support mixed blocks in bytecode.
- *)
+   normal blocks in bytecode. Hash raises on mixed blocks (but not
+   on normal blocks).
+
+   We could consider making native hash return a value instead. But,
+   if it's a different value than bytecode, it's important that users
+   can't marshal mixed block values from native code to bytecode.
+   (Otherwise you could marshal a hashtable, breaking its invariants.)
```

Comment:
(Not really about this PR, just to be
sure I understand.)
One can still marshal from bytecode
to native, and the unmarshaled value
will contain regular blocks rather than
mixed ones, which will pressumably
lead to unexpected results. Is that
correct?

Reply:
Incidentally, that probably means we
should never use mixed blocks in the
compiler (or at least `file_format/`
and dependencies).